src/handlers/indexers/stop_indexer.rs

```diff
@@ -34,3 +35,48 @@ pub async fn stop_indexer(
 
     Ok(())
 }
+
+pub async fn update_indexer_state(id: Uuid, new_status: IndexerStatus) -> Result<(), IndexerError> {
```

Comment:
this function is only be used in one place right ? 

Please add rustdoc to make sense of it, the `check_redundant_update_call` seems to just be checking that we are trying to update an indexer status to the same state it's already in, right ?